### PR TITLE
feat!: update flags for parca-agent 0.10.0

### PR DESCRIPTION
**Pull request by morremeyer**
This PR makes the chart compatible with parca-agent 0.10.0.

BREAKING CHANGE: This breaks compatibility with parca-agent < 0.10.0

### CODE REVIEW

File: charts/parca/README.md [hunk 1]
```diff
@@ -59,20 +65,20 @@ helm repo add parca https://parca-dev.github.io/helm-charts
 | agent.extraEnv | list | `[]` | Additional container environment variables for agent |
 | agent.image.pullPolicy | string | `"IfNotPresent"` | Overrides pullpolicy |
 | agent.image.repository | string | `"ghcr.io/parca-dev/parca-agent"` | Overrides the image repository |
-| agent.image.tag | string | `"v0.9.1"` | Overrides the image tag |
+| agent.image.tag | string | `"v0.10.0-rc.1"` | Overrides the image tag |
 | agent.logLevel | string | `"info"` | Agent log level |
 | agent.nodeSelector | object | `{}` | node selector for scheduling agent pods |
 | agent.podAnnotations | object | `{}` | Additional annotations for pods |
```
Comment (kakkoyun) [inline charts/parca/README.md:71]: There are also some removed flags such as this. We should clean up as well.

Comment (morremeyer) [reply charts/parca/README.md:71]: I removed the removed `--socket-path` flag.

**Comment by rlex** [comment]
Should we bump to 0.10.0 release and merge?

File: charts/parca/README.md (hunk 1 above)
Comment (kakkoyun) [reply charts/parca/README.md:72]: It was not just that one, unfortunately. There are several others. You can find the latest flags in here: https://github.com/parca-dev/parca-agent#configuration

**Review by kakkoyun — CHANGES_REQUESTED** [review]
We need to make sure, all the flags are up-to-date and bump the version.

**Comment by morremeyer** [comment]
`--pod-label-selector` is documented in https://github.com/parca-dev/parca-agent#kubernetes-label-selector, but does not exist anymore, it is therefore also removed.

**Comment by morremeyer** [comment]
I updated the image tag to `v0.10.0`, too. With this, this PR should be ready for merge.

I squashed all changes into once commit since all previous changes had updates and will need to be re-reviewed anyway.

**Review by metalmatze — APPROVED** [review]

**Comment by rlex** [comment]
probably worth bumping to 1.10.1 alraady

**Comment by morremeyer** [comment]
Good idea.

**Comment by morremeyer** [comment]
Good idea. Anything else to do here?

**Comment by rlex** [comment]
i think it's fine. CI is passing, too. Will merge now.

**Pull request merged**